tests/unit_tests/data_steward/retraction/create_deid_map_test.py

```diff
@@ -0,0 +1,173 @@
+
+    @classmethod
+    def setUpClass(cls):
+        print('**************************************************************')
+        print(cls.__name__)
+        print('**************************************************************')
+
+    def setUp(self):
+        self.project_id = 'test_project_id'
+        self.dataset_id = 'test_dataset_id'
+        self.all_datasets = [
+            'R2021q1r1_combined', 'R2021q1r1_deid', 'R2021q1r2_combined',
+            '2021q1r3_combined', 'R2021q1r3_deid'
+        ]
+        self.deid_datasets = ['R2021q1r1_deid', 'R2021q1r3_deid']
+        self.combined_datasets = [
+            'R2021q1r1_combined', 'R2021q1r2_combined', '2021q1r3_combined'
+        ]
+
+    @mock.patch('retraction.create_deid_map.utils.bq.list_datasets')
+    def test_get_combined_datasets_for_deid_map(self, mock_list_datasets):
+        result_all_datasets = self.all_datasets
+        result_deid_datasets = []
+        result_release_datasets = [
+            d for d in result_all_datasets
+            if consts.CURRENT_RELEASE_REGEX.match(d)
+        ]
+        for dataset in result_release_datasets:
+            if bool(re.match(create_deid_map.DEID_REGEX, dataset)) is True:
+                result_deid_datasets.append(dataset)
+        result_deid_and_combined_df = create_deid_map.get_corresponding_combined_dataset(
+            result_all_datasets, result_deid_datasets)
+
+        expected_all_datasets = mock_list_datasets.return_value = self.all_datasets
+        expected_deid_datasets = self.deid_datasets
+        expected_combined_datasets_df = create_deid_map.get_corresponding_combined_dataset(
+            expected_all_datasets, expected_deid_datasets)
+
+        pd.testing.assert_frame_equal(expected_combined_datasets_df,
+                                      result_deid_and_combined_df)
+
+    def test_get_corresponding_combined_dataset(self):
+        result = create_deid_map.get_corresponding_combined_dataset(
+            self.all_datasets, self.deid_datasets)
+        d = {
+            'deid_dataset': ['R2021q1r3_deid'],
+            'combined_dataset': ['2021q1r3_combined']
+        }
+        expected = pd.DataFrame(data=d)
+        pd.testing.assert_frame_equal(expected, result)
+
+    @mock.patch('utils.bq.get_table_info_for_dataset')
+    def test_check_if_deid_map_exists(self, mock_get_table_info):
+
+        expected_create_df = mock_get_table_info.return_value = pd.DataFrame(
+            data={
+                'table_name':
+                    ['_ehr_consent', common.PERSON, common.OBSERVATION]
+            })
+        expected_create_column_list = expected_create_df['table_name'].tolist()
+        if '_deid_map' in expected_create_column_list:
+            expected_create_return = consts.SKIP
+        elif 'deid_map' in expected_create_column_list:
+            expected_create_return = consts.RENAME
+        else:
+            expected_create_return = consts.CREATE
```

Comment:
Lol I think you might need to call the function `check_if_deid_map_exists` here to test it. Also you can get rid of all the `if` statements, since the function itself should be doing that, and replace them with just `self.assertEqual(result, consts.CREATE)` etc. since you know what to expect based on the input table names you supplied.

Reply:
lol oops! would I need to create multiple results then for each assertEqual?

Reply:
Not really, one should suffice